src/billing/Project.js

```diff
@@ -76,6 +71,25 @@ const ProjectDetail = ({
   const groups = groupByBillingAccountStatus(billingProject, workspacesInProject);
   const billingAccountsOutOfDate = !(_.isEmpty(groups.error) && _.isEmpty(groups.updating));
 
+  const signal = useCancellation();
+
+  const collectUserRoles = _.flow(
```

Comment:
Moved up from lower in the file because they need to be declared before referenced.